config/config.go

```diff
@@ -269,11 +309,9 @@ func (c *Config) extend(extensionConfig Config) {
 			}
 			baseRule.Tags = append(baseRule.Tags, currentRule.Tags...)
 			baseRule.Keywords = append(baseRule.Keywords, currentRule.Keywords...)
-			baseRule.Allowlist.Commits = append(baseRule.Allowlist.Commits, currentRule.Allowlist.Commits...)
```

Comment:
Why is this block getting deleted? The extend feature for a singular rule allowlist should function the same as before

Reply:
I believe this preserves the prior functionality, while allowing the new stuff to function as expected. (Correct me if I'm wrong.)

Consider the following config:
```toml
[extend]
useDefault = true

[[rules]]
id = "generic-api-key"
[rules.allowlist]
paths = ['''Some_Files\.xaml''']
stopwords = ["foo"]
```

With the append logic this would get translated as the following (pseudo-code):
```
allowlist{
  paths = ["some_files.xaml"]
  stopwords = ["foo", DefaultStopWords... ]
}
```

With the new logic, it would get translated as the following. Despite the difference in structure, the logic is still the same: `paths == "some_files.xaml" || stopwords == "foo" || stopwords == DefaultStopWords...`. 
```
allowlist{ // extended
  stopwords = [ DefaultStopWords... ]
},
allowlist{ // our config
  paths = ["some_files.xaml"]
  stopwords = ["foo" ]
}
```
---

As for the "new stuff", if either your or the extended rule's has an allowlist with `AND` then appending commits/paths/regexes/stopwords will break that.

It would be possible to carve out special handling for `[rules.allowlist]` vs. `[[rules.allowlist]]`, or append for `OR` and not `AND`, but I figured this way is the most straightforward. I can add more tests to be certain.

Reply:
> With the new logic, it would get translated as the following. Despite the difference in structure, the logic is still the same: paths == "some_files.xaml" || stopwords == "foo" || stopwords == DefaultStopWords....

I gotcha, that makes sense. Boy this would be much easier to test if I made that `config` command already